.github/local-actions/branch-manager/lib/main.ts

```diff
@@ -0,0 +1,161 @@
+import * as core from '@actions/core';
+import {PullRequestValidationConfig} from '../../../../ng-dev/pr/common/validation/validation-config.js';
+import {
+  assertValidPullRequestConfig,
+  PullRequestConfig,
+} from '../../../../ng-dev/pr/config/index.js';
+import {loadAndValidatePullRequest} from '../../../../ng-dev/pr/merge/pull-request.js';
+import {AutosquashMergeStrategy} from '../../../../ng-dev/pr/merge/strategies/autosquash-merge.js';
+import {
+  assertValidGithubConfig,
+  getConfig,
+  GithubConfig,
+  setConfig,
+} from '../../../../ng-dev/utils/config.js';
+import {AuthenticatedGitClient} from '../../../../ng-dev/utils/git/authenticated-git-client.js';
+import {
+  ANGULAR_ROBOT,
+  getAuthTokenFor,
+  revokeActiveInstallationToken,
+} from '../../../../github-actions/utils.js';
+import {MergeConflictsFatalError} from '../../../../ng-dev/pr/merge/failures.js';
+import {chdir} from 'process';
+import {spawnSync} from 'child_process';
+
+interface CommmitStatus {
+  state: 'pending' | 'error' | 'failure' | 'success';
+  description: string;
+}
+
+async function main(repo: {owner: string; repo: string}, token: string, pr: number) {
+  if (isNaN(pr)) {
+    core.setFailed('The provided pr value was not a number');
+    return;
```

Comment:
Nit: it is odd that the validation happens here while the type contract already explicitly is set to `number`

Reply:
`NaN` is a valid value for the type `number`, we could perform the check before calling it. But Its still "correct" to check it here.

Reply:
interesting. I expected number to not include NaN. Seems like this makes it very context-dependent.

Reply:
While typically anywhere you use a number you can use `NaN` right? it just propagates out all over.  So `2 * NaN` is `NaN`.

So it "functions" correctly as its not an error, but its definitely not ideal.

Reply:
Yes, but many applications perform an initial check- like here, but then down the line they only assume correct number operations (hence the context dependency and TS proposal for fine-grained separation here)

Reply:
I will move it outside of the function